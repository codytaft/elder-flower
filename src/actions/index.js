export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createCarer = user => ({
  type: 'CREATE_CARER',
  user
});

export const createElder = user => ({
  type: 'CREATE_ELDER',
  user
});
