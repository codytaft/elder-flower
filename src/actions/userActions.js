export const setCurrentUser = user => ({
  type: 'SET_CURRENT_USER',
  user
});

export const createUser = user => ({
  type: 'CREATE_USER',
  user
});
