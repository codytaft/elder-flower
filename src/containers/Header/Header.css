.main-header-section {
  display: flex;
  text-align: center;
  justify-content: space-between; }

.main-header-title {
  width: 100%; }

.nav-link-logout {
  display: flex;
  position: absolute;
  right: 1rem;
  justify-content: center;
  margin: auto;
  text-decoration: none;
  align-items: center;
  border: 0.1rem solid whitesmoke;
  border-radius: 1rem;
  height: 4rem;
  width: 10rem;
  font-size: 32px;
  color: whitesmoke; }

.nav-link-logout-hidden {
  display: none; }
