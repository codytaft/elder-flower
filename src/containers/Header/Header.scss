.main-header-section {
  display: flex;
  text-align: center;
  justify-content: space-between;
}

.main-header-title {
  // text-align: center;
  // display: flex;
  width: 100%;
  // margin: 0;
}

.nav-link-logout {
  display: flex;
  position: absolute;
  float: right;
  right: 1rem;
  justify-content: center;
  margin: auto;
  text-decoration: none;
  align-items: center;
  border: 0.1rem solid whitesmoke;
  border-radius: 1rem;
  height: 4rem;
  width: 10rem;
  font-size: 32px;
  color: whitesmoke;
}

.nav-link-logout-hidden {
  display: none;
}
