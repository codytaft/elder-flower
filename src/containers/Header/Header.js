import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link, withRouter } from 'react-router-dom';
import { setCurrentUser } from '../../actions';
import { connect } from 'react-redux';
import './Header.css';

export class Header extends Component {
  logoutUser = () => {
    const { setCurrentUser, history } = this.props;
    setCurrentUser(null);
    history.replace('/');
  };
  render() {
    const { currentUser } = this.props;
    return (
      <section className="main-header-section">
        {currentUser ? (
          <h1 className="main-header-title">Hello {currentUser.firstName}</h1>
        ) : (
          <h1 className="main-header-title">Welcome to Elder Flower </h1>
        )}
        <Link
          exact
          to="/"
          className={
            !this.props.currentUser
              ? 'nav-link-logout-hidden'
              : 'nav-link-logout'
          }
          onClick={this.logoutUser}
        >
          Logout
        </Link>
      </section>
    );
  }
}

Header.propTypes = {
  history: PropTypes.object,
  setCurrentUser: PropTypes.func,
  currentUser: PropTypes.object,
  location: PropTypes.object
};

export const mapStateToProps = state => ({
  currentUser: state.currentUser
});

export const mapDispatchToProps = dispatch => ({
  setCurrentUser: user => dispatch(setCurrentUser(user))
});

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Header)
);
