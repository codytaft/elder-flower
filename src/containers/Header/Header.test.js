import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { setCurrentUser } from '../../actions';

describe('Header', () => {
  let wrapper;
  let mockSetCurrentUser;
  let mockHistory;
  let mockCurrentUser;

  beforeEach(() => {
    mockSetCurrentUser = jest.fn();
    mockHistory = {
      length: 50,
      action: 'PUSH',
      location: { pathname: '/', search: '', hash: '', key: 'ur39zc' },
      replace: jest.fn()
    };
    mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };

    wrapper = shallow(
      <Header
        setCurrentUser={mockSetCurrentUser}
        history={mockHistory}
        currentUser={null}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot when currentUser exists', () => {
    wrapper = shallow(
      <Header
        setCurrentUser={mockSetCurrentUser}
        history={mockHistory}
        currentUser={mockCurrentUser}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should set setCurrentUser with null', () => {
    wrapper.instance().logoutUser();
    expect(mockSetCurrentUser).toHaveBeenCalledWith(null);
  });

  it('should set setCurrentUser with null', () => {
    wrapper.instance().logoutUser();
    expect(mockSetCurrentUser).toHaveBeenCalledWith(null);
  });
});

describe('mapStateToProps', () => {
  it('should return an object with a currentUser object', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockState = {
      currentUser: { ...mockCurrentUser }
    };

    const expected = {
      currentUser: {
        id: 75,
        firstName: 'Cody',
        lastName: 'Taft',
        phoneNumber: '+19038511575',
        email: 'cody.taft@gmail.com',
        password: '123',
        contactName: 'Gaynell',
        contactPhone: '+17203304593',
        isElder: false
      }
    };
    const mappedProps = mapStateToProps(mockState);
    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should invoke dispatch when using a function from mapDispatchToProps', () => {
    const mockCurrentUser = {
      id: 75,
      firstName: 'Cody',
      lastName: 'Taft',
      phoneNumber: '+19038511575',
      email: 'cody.taft@gmail.com',
      password: '123',
      contactName: 'Gaynell',
      contactPhone: '+17203304593',
      isElder: false
    };
    const mockDispatch = jest.fn();
    const actionToDispatch = setCurrentUser(mockCurrentUser);

    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setCurrentUser(mockCurrentUser);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
