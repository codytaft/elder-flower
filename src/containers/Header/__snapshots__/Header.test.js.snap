// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentUser={null}
    history={
        Object {
            "action": "PUSH",
            "length": 50,
            "location": Object {
              "hash": "",
              "key": "ur39zc",
              "pathname": "/",
              "search": "",
            },
            "replace": [Function],
          }
    }
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-header-title"
>
          Welcome to Elder Flower 
</h1>,
        <Link
          className="nav-link-logout-hidden"
          exact={true}
          onClick={[Function]}
          replace={false}
          to="/"
>
          Logout
</Link>,
      ],
      "className": "main-header-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Welcome to Elder Flower ",
          "className": "main-header-title",
        },
        "ref": null,
        "rendered": "Welcome to Elder Flower ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Logout",
          "className": "nav-link-logout-hidden",
          "exact": true,
          "onClick": [Function],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-header-title"
>
            Welcome to Elder Flower 
</h1>,
          <Link
            className="nav-link-logout-hidden"
            exact={true}
            onClick={[Function]}
            replace={false}
            to="/"
>
            Logout
</Link>,
        ],
        "className": "main-header-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Welcome to Elder Flower ",
            "className": "main-header-title",
          },
          "ref": null,
          "rendered": "Welcome to Elder Flower ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "className": "nav-link-logout-hidden",
            "exact": true,
            "onClick": [Function],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Header should match the snapshot when currentUser exists 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentUser={
        Object {
            "contactName": "Gaynell",
            "contactPhone": "+17203304593",
            "email": "cody.taft@gmail.com",
            "firstName": "Cody",
            "id": 75,
            "isElder": false,
            "lastName": "Taft",
            "password": "123",
            "phoneNumber": "+19038511575",
          }
    }
    history={
        Object {
            "action": "PUSH",
            "length": 50,
            "location": Object {
              "hash": "",
              "key": "ur39zc",
              "pathname": "/",
              "search": "",
            },
            "replace": [Function],
          }
    }
    setCurrentUser={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="main-header-title"
>
          Hello 
</h1>,
        <Link
          className="nav-link-logout"
          exact={true}
          onClick={[Function]}
          replace={false}
          to="/"
>
          Logout
</Link>,
      ],
      "className": "main-header-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Hello ",
            undefined,
          ],
          "className": "main-header-title",
        },
        "ref": null,
        "rendered": Array [
          "Hello ",
          undefined,
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Logout",
          "className": "nav-link-logout",
          "exact": true,
          "onClick": [Function],
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": "Logout",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="main-header-title"
>
            Hello 
</h1>,
          <Link
            className="nav-link-logout"
            exact={true}
            onClick={[Function]}
            replace={false}
            to="/"
>
            Logout
</Link>,
        ],
        "className": "main-header-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Hello ",
              undefined,
            ],
            "className": "main-header-title",
          },
          "ref": null,
          "rendered": Array [
            "Hello ",
            undefined,
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Logout",
            "className": "nav-link-logout",
            "exact": true,
            "onClick": [Function],
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": "Logout",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
